Базовый функционал для сервиса оценки и рекомендаций фильмов.

Solntmore opened the pull request:
Создан класс User, который определяет параметры пользователя.
Создан класс Film, который определяет параметры фильма.
Создан класс FilmController, который обрабатывает запросы показать все фильмы, добавить фильм и обновить информацию о фильмe.
Создан класс UserController, который обрабатывает запросы показать всех пользователей, добавить пользователя  и обновить информацию о пользователе.
Создан класс UserValidator, который проверяет корректность данных о пользователе.
Создан класс FilmValidator, который  проверяет корректность данных о фильме.
Создан класс IdCounter для реализация функционала присвоения ID.
Создана папка exceptions, в которой размещены класса кастомных исключений.

Solntmore committed:
Версия 1. Создан шаблон программы с помощью Spring Initializr.

Solntmore committed:
Версия 2. <feat> Созданы контролеры для обработки команд создания, обновления и получения списка всех фильмов, а также пользователей.
          <feat> Написаны тесты для контроллеров команд фильмов и пользователей.
          <feat> В контролеры добавлено логирование действия пользователя.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,57 @@
+
+    @GetMapping
+    public Collection<Film> findAll() {
+        log.debug("Получен запрос GET /films.");
```
Зачем два логирования подряд? Лучше было бы объединить в одно.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,65 @@
+@RequestMapping("/users")
+public class UserController {
+    private final Map<Integer, User> users = new HashMap<>();
+    UserValidator userValidator = new UserValidator();
```
Здесь лучше было бы добавить модификатор private.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,65 @@
+
+    @GetMapping
+    public Collection<User> findAll() {
+        log.debug("Получен запрос GET /users.");
```
Такое же пожелание: объединить два вызова в один.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/helpTools/IdCounter.java:
```diff
@@ -0,0 +1,13 @@
+
+@Data
+public class IdCounter {
+    int id;
```
Тоже самое: лучше бы добавить модификатор private,  тем более что getter и Setter  будут созданы.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,24 @@
+import java.time.LocalDate;
+
+@Data
+public class Film {
```
Для этих полей можно было бы добавить аннотации:
@notblank
@SiZe(min = 1, max = 200)
@NotNull
@min(1)

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,27 @@
+    @Email(message = "Email должен быть корректным адресом электронной почты.")
+    private String email;
+    @NotBlank(message = "Необходимо указать login.")
+    private String login;
```
Поле login
Сюда хорошо бы добавить аннотацию.
@pattern(regexp = "^\\\S*$") 
нужно, чтобы пробелов вообще не было.

avfyodorov commented on src/test/java/ru/yandex/practicum/filmorate/controller/FilmControllerTest.java:
```diff
@@ -0,0 +1,197 @@
+@WebMvcTest
+public class FilmControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
```
Этого по программе вы ещё не проходили, насколько я помню. Значит, сами разобрались. Здорово!

Solntmore replied:
Ждет нас в 15 спринте! Но к этой теме очень подходит.

avfyodorov reviewed (commented):
Добрый вечер!
Работа получилась очень хорошей, серьёзных замечаний нет. Есть мелкие поправки и только.

Solntmore committed:
Версия 3. <refactor> Блоки логирования в классе UserController и FilmController объединены с сохранением прежнего функционала.
          <refactor> Добавлены приватные модификаторы в классах UserController и IdCounter.
          <refactor> Добавлены дополнительные аннотации, проверяющие корректность ввода данных для параметров в классе UserController и FilmController.

avfyodorov reviewed (commented):
Добрый день!

Замечаний нет.
Работа принята.
